homeassistant/helpers/intent.py

```diff
@@ -438,6 +439,30 @@ def _filter_by_name(
                     break
 
 
+def _matches_name(
+    state: State,
+    name: str,
+    entities: entity_registry.EntityRegistry,
+) -> bool:
+    """Return true if entity's (normalized) name or alias matches the provided name."""
+    name_norm = _normalize_name(name)
+
+    if (state.entity_id == name) or _normalize_name(state.name) == name_norm:
```

Comment:
We call `_normalize_name` a lot. I don't know if it's worth doing some caching ?

Reply:
I wonder is using this normalization function would work: https://github.com/home-assistant/core/blob/e30db943db977aae157971d28b54557e2f2ccd3d/homeassistant/helpers/normalized_name_base_registry.py#L27

It ignores all whitespace, though.

Reply:
I can see ignoring whitespace in some very specific scenarios, such as "overhead light" vs. "over head light". But it should probably not be the default.